swirl_proc_macro/src/background_job.rs

```diff
@@ -0,0 +1,209 @@
+#![warn(warnings)]
+use crate::diagnostic_shim::*;
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::punctuated::Punctuated;
+use syn::spanned::Spanned;
+
+pub fn expand(item: syn::ItemFn) -> Result<TokenStream, Diagnostic> {
```

Comment:
I get why all of this needs to be this complicated, but whoa! it's a lot to take in 😆